### ISSUE TITLE
Make using local repos easier/more intuitive

### ISSUE THREAD

**Issue by jsingle** [gamechanger/dusty#531]
This has been a source of confusion for a couple new users, most recently @melkorm.  If you specify a local path for your repo in an app or lib spec, Dusty still checks it out and manages it internally.  The more intuitive behavior would be for Dusty to treat that repo as if it has a "local override".

It might make sense for any repo which is just specified as a local path to behave as a user managed repo. If the desired behavior is for dusty to `git pull` from a local path, then a user could still specify `file://` at the front of the path, so we could maintain that functionality

**Comment by thieman** [comment]
I agree that any repo specified with a local path should be treated as overwritten and always source from the local path the user gave us.

I don't think we need to fork behavior based on whether the path was specified with `file://` or not, that seems magical and unintuitive.

**Comment by jsingle** [comment]
So there's no good reason for the current behavior regarding to exist?  No usecases that you can think of where you'd want dusty to `git clone` from a local filepath?

**Comment by torbjornvatn** [comment]
:+1: for this feature, and when I'm at it I'd like the possibility to use environment variables in this local repos definitions as well. 
It would be useful when an entire team shares the Dusty specs, but naturally has the local repos checked out at different absolute file paths. 
E.g. `repo: $GREAT_PROJECT_HOME/great-project-app-1`

**Comment by thieman** [comment]
When we do this, let's also make sure that `dusty repos list` shows local repos as overridden. cc @nrschultz
